views/assets/js/phone_number_field.js

```diff
@@ -35,41 +35,44 @@ class PhoneNumberField {
             showFlags: false,
             showSelectedDialCode: true,
             initialCountry: this.element.dataset.defaultCountry,
-            defaultToFirstCountry: Boolean(this.element.dataset.defaultCountry)
+            defaultToFirstCountry: Boolean(this.element.dataset.defaultCountry),
+            useFullscreenPopup: this.isMobile
         })
 
         // need to adjust the MDC floating label's left offset to account for the country code
         // picker:
-        this.input.addEventListener("countrychange", (event) => {
-            this.updateDropdown()
-        })
-
-        this.input.addEventListener("input", (event) => {
-            this.updateDropdown()
-        })
-
-        this.input.addEventListener("blur", (event) => {
-            this.updateDropdown()
-        })
-
-        this.input.addEventListener("focus", (event) => {
-            this.updateDropdown()
-        })
+        this.input.addEventListener("countrychange", () => this.updateDropdown())
+        this.input.addEventListener("input", () => this.updateDropdown())
+        this.input.addEventListener("blur", () => this.updateDropdown())
+        this.input.addEventListener("focus", () => this.updateDropdown())
```

Comment:
I think you could just do this:

```suggestion
        this.input.addEventListener("countrychange", this.updateDropdown)
        this.input.addEventListener("input", this.updateDropdown)
        this.input.addEventListener("blur", this.updateDropdown)
        this.input.addEventListener("focus", this.updateDropdown)
```

Reply:
that form causes the handler function to be bound to the element that emitted the event, so any references to `this` would blow up expecting an instance of the plugin class. the alternative is e.g. `this.input.addEventListener("focus", this.updateDropdown.bind(this))`, which (imo) isn't quite as clear for people not familiar with JS quirks.

Reply:
I think the intent might be clearer that way. I've always struggled to know what `this` is at any given moment in JS.